v2/internal/controller/controllers/targetnamespace_test.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package controllers_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/microsoft.resources/v1alpha1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/controller/controllers"
+	"github.com/Azure/azure-service-operator/v2/internal/controller/reconcilers"
+	"github.com/Azure/azure-service-operator/v2/internal/controller/testcommon"
+
+	. "github.com/onsi/gomega"
+)
+
+const finalizerName = reconcilers.GenericControllerFinalizer
+
+const (
+	timeoutFast = time.Minute * 3
```

Comment:
This seems like a really long fast timeout, especially for recording tests. We should craft the recording tests in such a way that they're _really_ fast, but here there are cases/modes where we're waiting for 3m no matter what?

Should this be making use of `tc.RemainingTime()`? There's also already a `tc.PollingInterval()` which is also optimized for speed in replaying.

Reply:
Good point, I'll check this.

Reply:
Rejigging the tests to pass in the config ended up getting rid of these, since I switched everything to just using `tc.Eventually`.